test/extension-tests/successful-build/test/prefered-generators.test.ts

```diff
@@ -0,0 +1,213 @@
+import * as chai from 'chai';
+import {expect} from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import {clearExistingKitConfigurationFile} from '../../../test-helpers';
+import {DefaultEnvironment} from '../../../helpers/test/default-environment';
+
+import {CMakeTools} from '../../../../src/cmake-tools';
+import {ITestCallbackContext, IHookCallbackContext} from 'mocha';
+
+interface BuildSystemConfiguration {
```

Comment:
Might be better named `KitEnvironment` or something

Reply:
Done